# COGS 108 - Data Checkpoint

## Authors
Team List:
- Marcus de Ramos: Project Administration, Conceptualization, Analysis, Writing - original draft, Writing - review & editing
- Roberto Echeverria Sosa: Background Research, Writing - original draft, Writing - review & editing
- Lucas Johnson: Background Research, Data curation, Writing - original draft, Writing - review & editing
- Yuki Saeli: Data collection, Data curation, Writing – original draft, Writing - review & editing
- Andrew Zhao: Background Research, Writing - review & editing, Conceptualization, Methodology

## Research Question

How does movie runtime relate to critic scores (Rotten Tomatoes) and audience scores (IMDb rating) independently, and does that relationship differ by age rating or decade?


## Background and Prior Work

### Introduction:
We chose our topic because we wanted to explore data on the entertainment industry. Is art truly subjective, or can we point to certain characteristics that show us when a show/movie will have a large rating gap between critics and viewers. By exploring media data we can see different factors that contribute to ratings and we may uncover traits that make the critics and viewers disagree. 

From the project the Stanford University students conducted, they only got a ~ 52% prediction accuracy. They mentioned this was due to the fact that people that leave movie comments usually rate in extremes. If the movie was a 7, they’d rate it a 3-4 to bring the average down and vice versa. 

In the project from the University of Malta, they used film script to predict audience ratings and were more successful. The model's prediction was off by 1 point of the actual rating 84% of the time, and the classifier placed the film in the correct bucket 83% of the time.

In the analysis on IMDb ratings conducted by Carnegie Mellon University students, one discovery made was that movie content ratings and movie title affected IMDb scores depending on the decade the movie was released and could be used to predict the scores of upcoming movies. Further, they investigate how release season correlates with IMDb scores and profits.

Though many pieces of work explore rating prediction using several methods, none explore the characteristics that make a movie polarizing. That is what we want to dive deeper into, what makes a movie’s rating ambiguous? And how do we quantify that?

### Explanation of work already done: 
A similar project was conducted by using movie reviews to predict the amount of stars it got. This relates to our project because it ties in personal feedback with overall rating. This project used Naive Bayes to rate the movies/shows and was from students at Stanford University.

There was another project done by Karl Cini and John Abela from the Department of Computer Information Systems, Faculty of ICT, University of Malta, Msida Campus, Msida, MSD 2080, Malta. They read in script data and used NLP (Natural Language Processing) to predict film audience ratings. By testing out several methods such as XGBoost and MLPS, they found that SVR performed the best on unseen data and was stable results wise. This relates to our project by using how certain film characteristics can influence audience reviews. These characteristics can have strong overlap with what film critics look for. The characteristics that lead to better ratings may be subconsciously affecting audiences to score certain ways.

The last relevant project discussed is IMDb Movie Analysis by Josh Braverman, Pranav Reddy, and Seonho Park, who are students at Carnegie Mellon University. By scraping data from IMDb’s webpage extracting 15 qualitative and quantitative variables to understand trends and  answer their research questions. From the data gathered, they discovered the impact of age ratings on potential success of movies, how movie titles have evolved over the decades, as well as the impact a movie's release date has on IMDb scores and profits. This study is directly relevant to our project as it uses information from IMDb to form conclusions about the impacts of movie characteristics, which is essentially what our project will be doing. 

### Citations:
1. Mehta, P., & Scaria, S. (2011). Predicting Star Ratings from Movie Review Comments. Stanford University CS229. https://cs229.stanford.edu/proj2011/MehtaPhilipScaria-Predicting%20Star%20Ratings%20from%20Movie%20Review%20Comments.pdf
2. Xie, W. (2025). Predicting Audience Scores Based On Film Script. ScienceDirect / Entertainment Computing. https://www.sciencedirect.com/science/article/pii/S1875952125001235#sec4
3. Team 24. (2022). IMDb Movie Analysis. Carnegie Mellon University Statistics & Data Science. https://www.stat.cmu.edu/capstoneresearch/fall2022/315files_f22/team24.html

## Hypothesis


We predict that movies with runtimes that are longer than the average movie runtime will receive lower ratings from a general audience while critic reviews will be unaffected. We believe this because movies with longer runtimes may lose the interest of a general audience, while critics are more likely to base their ratings on other aspects of the movie. Further, we anticipate that for movies with lower age ratings(7+), a lower duration will result in higher ratings across both IMDB and Rotten Tomatoes as younger audiences have shorter attention spans. Lastly, we expect that movies with higher ratings before the 2000's will have longer runtimes than movies after the 2000's due to a change in audience attention span.

## Data

In [1]:
# Setup code -- this only needs to be run once after cloning the repo!
# this code downloads the data from its source to the `data/00-raw/` directory
# if the data hasn't updated you don't need to do this again!

# if you don't already have these packages (you should!) uncomment this line
# %pip install requests tqdm

# import sys
# sys.path.append('./modules') # this tells python where to look for modules to import

# import get_data # this is where we get the function we need to download data

# replace the urls and filenames in this list with your actual datafiles
# yes you can use Google drive share links or whatever
# format is a list of dictionaries; 
# each dict has keys of 
#   'url' where the resource is located
#   'filename' for the local filename where it will be stored 
# datafiles = [
#    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/airline-safety/airline-safety.csv', 'filename':'airline-safety.csv'},
#    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/bad-drivers/bad-drivers.csv', 'filename':'bad-drivers.csv'}
#]

#get_data.get_raw(datafiles,destination_directory='data/00-raw/')

## Combined Streaming Platform Movie Dataset
### Data overview


  - Dataset Name: Combined Streaming Platform Movie Dataset
  - Link to datasets:
    The merged dataset was created by combining the following sources:

    - [Movies on Netflix, Prime Video, Hulu and Disney+](https://www.kaggle.com/datasets/ruchi798/movies-on-netflix-prime-video-hulu-and-disney)

    - [Netflix Titles Dataset](https://www.kaggle.com/datasets/shivamb/netflix-shows)

    - [Disney+ Titles Dataset](https://www.kaggle.com/datasets/shivamb/disney-movies-and-tv-shows)

     - [Amazon Prime Titles Dataset](https://www.kaggle.com/datasets/shivamb/amazon-prime-movies-and-tv-shows)

    - [Hulu Titles Dataset](https://www.kaggle.com/datasets/shivamb/hulu-movies-and-tv-shows)


  - Number of observations: The original source datasets contained several thousand entries combined. After merging datasets, filtering for movies, removing duplicates, and dropping missing values, the final cleaned dataset used for analysis contains 1,713 movie entries.

  - Number of variables: 6 variables in the final processed dataset

  ### Relevant variables:

- `Title` — movie title used for merging datasets
- `IMDB_Rating` — audience rating measured on a 0–10 numerical scale
- `Rotten Tomatoes` — critic score measured on a 0–100 scale
- `Runtime` — movie runtime measured in minutes
- `Age_Rating` — standardized movie age rating category
- `Year` — release year of the movie


### Dataset Description
For this project, we combined multiple movie datasets from streaming platforms and public movie review databases into one larger dataset. Since no single dataset included all the variables needed for our research question, we merged datasets containing movie runtimes, audience ratings, critic scores, release years, and age ratings. The final dataset was created by combining Netflix, Hulu, Amazon Prime, and Disney+ movie data with IMDb ratings and Rotten Tomatoes critic scores.

### Important Metrics
One of the most important variables in this dataset is runtime, which is measured in minutes. Runtime represents the total length of a movie and is the main variable used to test our hypothesis. Most movies in the dataset fall between around 80 and 120 minutes which is considered standard feature film length, although some are much shorter or longer. IMDb ratings are measured on a 0–10 scale and reflect audience opinions from users who leave ratings online. In general, movies with ratings above 7 are considered positively received by audiences, while ratings below 5 are usually viewed more negatively. Rotten Tomatoes scores are measured on a 0–100 scale and represent critic approval percentages. Higher Rotten Tomatoes scores generally indicate stronger critic reception.   


### Dataset Limitations
There are a few concerns with our dataset that could affect how well it tests our hypothesis. The biggest issue is that both rating columns come from self-selected groups rather than a true general audience. IMDB ratings come from users who chose to rate a film online, which skews younger, more male, and toward people who are already fans of the movie they are rating. People who started a long movie, got bored, and turned it off are exactly the viewers our hypothesis cares about, and they are the least likely to show up in the IMDB sample. This may bias the data against finding the effect we predicted. Rotten Tomatoes scores come from accredited critics, mostly covering US and UK theatrical releases, which is a different kind of bias and tends to favor prestige films that often happen to be longer. 
   
Another concern is that runtime in this dataset is confounded with content type. G rated films average 82 minutes while PG-13 films average 106 minutes, and our dataset includes 125 films under 60 minutes that are likely stand-up specials and short documentaries rather than feature films. This pulls the average runtime down to 98 minutes, which is shorter than what most people would consider a normal feature length, so our split between long and short movies may end up partly being a split between features and short-form content.

The dataset is also heavily skewed toward recent years, with about 67 percent of films from 2015 onward and only 121 films from before 1980, and the older films that did make the list are survivors that stayed culturally relevant, which inflates their ratings on both sides. We also do not have rating counts, so a film with 12 IMDB ratings is treated the same as one with 800,000. These biases are worth keeping in mind as we move into the analysis, since they could either mask or exaggerate the effect we are looking for.


### Load the Dataset


We initially had quite a few data sets that we were looking at. However since we are discussing the relationship between Movie runtimes, IMDB score, Rotten Tomato Score, the decade they were made, and their age ratings, there was no one data set that had all of these variables. To solve this problem we decided to combine all of these data sets.

Firstly we need to get all our data from each of the platforms into one big data set.

In [2]:
import pandas as pd
import numpy as np

netflix = pd.read_csv('data/00-raw/netflix_titles.csv')
hulu = pd.read_csv('data/00-raw/hulu_titles.csv')
prime = pd.read_csv('data/00-raw/amazon_prime_titles.csv')
disney = pd.read_csv('data/00-raw/disney_plus_titles.csv')

# Combine
platforms = pd.concat([netflix, hulu, prime, disney], ignore_index=True)

Next We will filter the data and only keep the movie entries

In [3]:
# Filter for Movies
movies_platforms = platforms[platforms['type'] == 'Movie'].copy()

movies_platforms = movies_platforms[['title', 'duration', 'rating', 'release_year']]
movies_platforms.rename(columns={'duration': 'Runtime', 'rating': 'Age_Rating_Platform', 'release_year': 'Year_Platform'}, inplace=True)

# Drop duplicates based on title
movies_platforms.drop_duplicates(subset=['title'], inplace=True)

Next we need to add the IMDB and Rotten tomato ratings to each movie

In [4]:
# Load MoviesOnStreamingPlatforms
movies_sp = pd.read_csv('data/00-raw/MoviesOnStreamingPlatforms.csv')
movies_sp = movies_sp[['Title', 'Year', 'Age', 'Rotten Tomatoes']]

# Load IMDB
imdb = pd.read_csv('data/00-raw/imdb_ratings.csv')
imdb = imdb[['title', 'averageRating']]
imdb.rename(columns={'averageRating': 'IMDB_Rating', 'title': 'Title'}, inplace=True)

# Drop duplicates based on title for IMDB
imdb.drop_duplicates(subset=['Title'], inplace=True)

# Merge
merged = pd.merge(movies_sp, imdb, on='Title', how='inner')
merged = pd.merge(merged, movies_platforms, left_on='Title', right_on='title', how='inner')

# Combine Age and Year
merged['Final_Age'] = merged['Age'].fillna(merged['Age_Rating_Platform'])
merged['Final_Year'] = merged['Year'].fillna(merged['Year_Platform'])

# Select required columns
final_df = merged[['Title', 'IMDB_Rating', 'Rotten Tomatoes', 'Runtime', 'Final_Age', 'Final_Year']].copy()
final_df.rename(columns={'Final_Age': 'Age_Rating', 'Final_Year': 'Year'}, inplace=True)

Now that we have everything merged, we can clean our data by dropping any entries with null values

In [5]:
final_df_clean = final_df.dropna(subset=['IMDB_Rating', 'Rotten Tomatoes', 'Runtime', 'Age_Rating', 'Year'])

print(f"Final dataset shape: {final_df_clean.shape}")
print(final_df_clean.head())

merged_data_path = 'data/01-interim/combined_movies_data.csv'
final_df_clean.to_csv(merged_data_path, index=False)

Final dataset shape: (1713, 6)
                        Title  IMDB_Rating Rotten Tomatoes  Runtime  \
0                The Irishman          7.9          98/100  209 min   
1          The Social Dilemma          7.7          93/100   94 min   
2  The Trial of the Chicago 7          7.8          92/100  130 min   
3         Dolemite Is My Name          7.3          92/100  118 min   
4              Miss Americana          7.4          90/100   85 min   

  Age_Rating  Year  
0        18+  2019  
1        13+  2020  
2        18+  2020  
3        18+  2019  
4        18+  2020  


Now that everything is merged, there is still a few issues with our data set. Namely the ratings column is not standardized. In order to perform proper analysis, we need to standardize that column. Our mapping will be the following:

General Audiences
- 'G': 'G', 'TV-G': 'G', 'TV-Y': 'G', 'TV-Y7': 'G', 'TV-Y7-FV': 'G', 'all': 'G', 'ALL': 'G'

Parental Guidance Suggested
- 'PG': 'PG', 'TV-PG': 'PG', '7+': 'PG'

Teens
- 'PG-13': 'PG-13', 'TV-14': 'PG-13', '13+': 'PG-13', '16+': 'PG-13'

Restricted / Mature
- 'R': 'R', 'TV-MA': 'R', '18+': 'R', 'NC-17': 'R'

Not Rated
- 'NR': 'NR', 'UR': 'NR', 'UNRATED': 'NR', 'Not Rated': 'NR'

In [6]:
# Load interim
df = pd.read_csv(merged_data_path)

rating_mapping = {
    'G': 'G', 'TV-G': 'G', 'TV-Y': 'G', 'TV-Y7': 'G', 'TV-Y7-FV': 'G', 'all': 'G', 'ALL': 'G',
    'PG': 'PG', 'TV-PG': 'PG', '7+': 'PG',
    'PG-13': 'PG-13', 'TV-14': 'PG-13', '13+': 'PG-13', '16+': 'PG-13',
    'R': 'R', 'TV-MA': 'R', '18+': 'R', 'NC-17': 'R',
    'NR': 'NR', 'UR': 'NR', 'UNRATED': 'NR', 'Not Rated': 'NR'
}

df['Age_Rating'] = df['Age_Rating'].replace(rating_mapping)

print("Standardized Age Ratings:")
print(df['Age_Rating'].value_counts())

Standardized_ratings = 'data/01-interim/combined_movies_data_standardized.csv'
df.to_csv(Standardized_ratings, index=False)

Standardized Age Ratings:
Age_Rating
R        721
PG-13    513
PG       296
G        179
NR         4
Name: count, dtype: int64


With our data set merged we can now finally load it

In [7]:

df = pd.read_csv(Standardized_ratings)
display(df.head())

,Title,IMDB_Rating,Rotten Tomatoes,Runtime,Age_Rating,Year
0,The Irishman,7.9,98/100,209 min,R,2019
1,The Social Dilemma,7.7,93/100,94 min,PG-13,2020
2,The Trial of the Chicago 7,7.8,92/100,130 min,R,2020
3,Dolemite Is My Name,7.3,92/100,118 min,R,2019
4,Miss Americana,7.4,90/100,85 min,R,2020


**Is the dataset tidy?**
Yes, the dataset is mostly tidy because it adheres to the principles of tidy data:
1. Each variable forms a column (Title, IMDB_Rating, Rotten Tomatoes, Runtime, Age_Rating, Year).
2. Each observation forms a row (one row per movie).


However, two of our columns (`Rotten Tomatoes` and `Runtime`) contain mixed numeric and string characters ("98/100" and "209 min"). While structurally tidy, these need to be converted to pure numeric data types for analysis.

**Dataset Size:**
We can verify the size of the dataset using the `.shape` and `.info()` methods below.

In [8]:
print(f"The dataset contains {df.shape[0]} rows (observations) and {df.shape[1]} columns (variables).")
print("-" * 50)
df.info()

The dataset contains 1713 rows (observations) and 6 columns (variables).
--------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1713 entries, 0 to 1712
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Title            1713 non-null   object 
 1   IMDB_Rating      1713 non-null   float64
 2   Rotten Tomatoes  1713 non-null   object 
 3   Runtime          1713 non-null   object 
 4   Age_Rating       1713 non-null   object 
 5   Year             1713 non-null   int64  
dtypes: float64(1), int64(1), object(4)
memory usage: 80.4+ KB


**Missing Data Analysis**
We will check for missing values using `.isna().sum()`.

In [9]:
missing_data = df.isna().sum()
print("Missing data per column:")
print(missing_data)

Missing data per column:
Title              0
IMDB_Rating        0
Rotten Tomatoes    0
Runtime            0
Age_Rating         0
Year               0
dtype: int64


**Data Cleaning**
Before checking for outliers, we firstly need to convert the `Rotten Tomatoes` and `Runtime` columns into a numerical format. We will strip the strings "/100" and " min" from the text and convert them to integers.

In [10]:
df.to_csv('data/01-interim/combined_movies_data_interim.csv', index=False)

#Converting 'Rotten Tomatoes' to integers
df['Rotten Tomatoes'] = df['Rotten Tomatoes'].str.replace('/100', '').astype(int)

#Convert 'Runtime' to integers
df['Runtime'] = df['Runtime'].str.replace(' min', '').astype(int)

display(df.describe())

,IMDB_Rating,Rotten Tomatoes,Runtime,Year
count,1713.000000,1713.000000,1713.000000,1713.000000
mean,6.252715,56.478109,98.186807,2010.181553
std,1.248431,12.693167,28.307064,17.702509
min,1.500000,12.000000,1.000000,1920.000000
25%,5.500000,47.000000,87.000000,2010.000000
50%,6.400000,55.000000,97.000000,2018.000000
75%,7.200000,65.000000,112.000000,2019.000000
max,9.600000,98.000000,312.000000,2021.000000


**Outliers and Suspicious Entries**
Now that we have converted the `Rotten Tomatoes` and `Runtime` columns into a numerical format we can check for outliers

* **Year:** Ranges from 1920 to 2021. There are no negative or futuristic years.
* **IMDb Ratings:** Range from 1.5 to 9.6, which falls perfectly within the 0-10 scale.
* **Runtime:** We will look for 0-minute runtimes or highly unusual durations (e.g., massive outliers). 
* **Rotten Tomatoes:** We will verify the range is strictly between 0 and 100.

In [11]:
suspicious_runtimes = df[(df['Runtime'] < 10) | (df['Runtime'] > 300)]
if not suspicious_runtimes.empty:
    print("Found suspicious runtimes:")
    display(suspicious_runtimes.head())
else:
    print("No highly suspicious runtimes found out of bounds.")

final_path = 'data/02-processed/combined_movies_data_processed.csv'
df.to_csv(final_path, index=False)
print(f"\nFully wrangled data successfully saved to: {final_path}")

Found suspicious runtimes:


,Title,IMDB_Rating,Rotten Tomatoes,Runtime,Age_Rating,Year
84,Black Mirror: Bandersnatch,7.2,75,312,R,2018
353,Canvas,6.6,57,9,G,2020
556,Cops and Robbers,7.3,50,8,PG-13,2020
662,Sol Levante,4.7,46,5,PG-13,2020
1003,The Hug,6.1,52,5,PG-13,2018



Fully wrangled data successfully saved to: data/02-processed/combined_movies_data_processed.csv


## Ethics

### A. Data Collection
 - [X] **A.1 Informed consent**: If there are human subjects, have they given informed consent, where subjects affirmatively opt-in and have a clear understanding of the data uses to which they consent?

    * This project uses publicly available movie and television metadata and ratings from existing datasets. No humans are directly involved in data collection, so informed consent is not applicable

 - [X] **A.2 Collection bias**: Have we considered sources of bias that could be introduced during data collection and survey design and taken steps to mitigate those?

    * This dataset may be biased because it includes more popular movies, major streaming platforms, and highly reviewed titles, while less known movies, indie films, and international content may not be included as much. Because of this, the data may not represent the full film industry. This could lead our results to favor already popular studios and films instead of showing a balanced view.


 - [X] **A.3 Limit PII exposure**: Have we considered ways to minimize exposure of personally identifiable information (PII) for example through anonymization or not collecting information that isn't relevant for analysis?

    * No personally identifiable information (PII) is being collected or used in this project. The dataset only includes public information like title names, ratings, genres, release year, and streaming platform availability.

 - [X] **A.4 Downstream bias mitigation**: Have we considered ways to enable testing downstream results for biased outcomes (e.g., collecting data on protected group status like race or gender)?

    * We will keep this bias in mind when doing our analysis and avoid making broad claims about all movies. We understand that some groups, like independent filmmakers or underrepresented communities, may not be fully represented in the dataset. Because of this, we will clearly state that our results only reflect the data we are using.

### B. Data Storage
 - [X] **B.1 Data security**: Do we have a plan to protect and secure data (e.g., encryption at rest and in transit, access controls on internal users and third parties, access logs, and up-to-date software)?

     * Our project uses public movie data and review scores, so we are not storing any personal or private information. The files are only shared between our group members, which helps keep the data organized and prevents unwanted changes.

 - [X] **B.2 Right to be forgotten**: Do we have a mechanism through which an individual can request their personal information be removed?

     * Since our dataset does not include any personal information, there is no need for a process to remove someone’s data. We are only using public movie information like runtime, ratings, genre, and release details.

 - [X] **B.3 Data retention plan**: Is there a schedule or plan to delete the data after it is no longer needed?

     * We will keep this dataset for the duration of our project and possibly for future reference in class. Since the data is public and does not contain private information, keeping it does not raise major ethical concerns.

### C. Analysis
 - [X] **C.1 Missing perspectives**: Have we sought to address blindspots in the analysis through engagement with relevant stakeholders (e.g., checking assumptions and discussing implications with affected communities and subject matter experts)?
      
      * Our project mainly uses reviews from IMDb and Rotten Tomatoes, which mostly come from critics and active reviewers. This means we are missing perspectives from general audiences who don’t leave reviews, as well as filmmakers from underrepresented backgrounds whose work may not get as much attention. Because of this, our analysis may not fully represent all viewpoints and may lean toward more mainstream opinions.

 - [X] **C.2 Dataset bias**: Have we examined the data for possible sources of bias and taken steps to mitigate or address these biases (e.g., stereotype perpetuation, confirmation bias, imbalanced classes, or omitted confounding variables)?

      * Since we are only using data from these two sites, our data reflects a specific group of people like critics and heavy movie watchers. These groups may have different opinions compared to the general public. Also, people are more likely to leave reviews when they strongly like or dislike a movie, which can skew the data. We will keep this in mind and mention it when interpreting our results.

 - [X] **C.3 Honest representation**: Are our visualizations, summary statistics, and reports designed to honestly represent the underlying data?
      * We will present our visualizations, statistics, and reports to honestly represent the data. 

 - [X] **C.4 Privacy in analysis**: Have we ensured that data with PII are not used or displayed unless necessary for the analysis?
      * No personally identifiable information (PII) is being collected or used in this project. The dataset only includes public information like title names, ratings, genres, release year, and streaming platform availability.

 - [X] **C.5 Auditability**: Is the process of generating the analysis well documented and reproducible if we discover issues in the future?

      * We will take measures to make the analysis reproducable by providing the code and dataset necessary for reproduction. 
      
### D. Modeling
 - [X] **D.1 Proxy discrimination**: Have we ensured that the model does not rely on variables or proxies for variables that are unfairly discriminatory?
     * Even though our model focuses on movies, it can still learn patterns that reflect real-world bias, like favoring certain genres, countries, or studios. This could indirectly disadvantage films from less represented groups. We will be careful when interpreting our results and avoid making claims that could reinforce these biases.
 - [X] **D.2 Fairness across groups**: Have we tested model results for fairness with respect to different affected groups (e.g., tested for disparate error rates)?
     * Even if we are not directly analyzing people, our results can still affect groups like filmmakers and studios. If the model works better for mainstream movies, it could reinforce the idea that those movies are “better” while overlooking others. We will check if certain types of films are being favored and clearly explain this as a limitation.
 - [X] **D.3 Metric selection**: Have we considered the effects of optimizing for our defined metrics and considered additional metrics?
     * The only effects we've spoken about right now would be having better movie rating models that factor in never before 
       analyzed factors.
 - [X] **D.4 Explainability**: Can we explain in understandable terms a decision the model made in cases where a justification is needed?
     * Yes, our model will have clear defined parameters and logic that will be easily explainable.
 - [X] **D.5 Communicate limitations**: Have we communicated the shortcomings, limitations, and biases of the model to relevant stakeholders in ways that can be generally understood?
     * Yes, we understand that all aspects of movie/show data would be costly in terms of compute and that we must analyze 
       trade offs in the features we decide to use.

### E. Deployment
 - [X] **E.1 Monitoring and evaluation**: Do we have a clear plan to monitor the model and its impacts after it is deployed (e.g., performance monitoring, regular audit of sample predictions, human review of high-stakes decisions, reviewing downstream impacts of errors or low-confidence decisions, testing for concept drift)?
 
    * Since this is a class project and not a real deployed model, ongoing monitoring is not needed. We will still review our results carefully and make sure our conclusions make sense based on the data.

 - [X] **E.2 Redress**: Have we discussed with our organization a plan for response if users are harmed by the results (e.g., how does the data science team evaluate these cases and update analysis and models to prevent future harm)?

    * Even though this is just a class project, biased or misleading results could still influence how people think about movies. If we notice issues in our analysis, we will fix them and update our results. We will also be clear about the limitations so people don’t misinterpret our findings.

 - [X] **E.3 Roll back**: Is there a way to turn off or roll back the model in production if necessary?

    * Since this project is not being deployed, there is nothing to roll back. If we make an error in our analysis or code, we can correct it and update our project files.

 - [X] **E.4 Unintended use**: Have we taken steps to identify and prevent unintended uses and abuse of the model and do we have a plan to monitor these once the model is deployed?

    * One possible issue is that people might think our model gives an objective answer for which movies are “better,” when it is actually based on biased and incomplete data. This could disadvantage movies that are not well represented in the dataset. We will make it clear that our results are limited and should not be taken as final judgments.



## Team Expectations 

* Meet once a week (In-Person or Zoom) to discuss progress, and next steps.
* Communicate regularly through Discord and respond within 24 hours.
* Finish assigned tasks on time so project stays on schedule.
* Contribute fairly by helping with coding, analysis, writing and project discussion.
* Address problems at least 2 days in advance by communicating if someone is falling behind so the team can adjust resposibilities. 
    - Responsibilities and workload will be shared amongst other team members in order to complete assignment in a timely manner, based on group discussion
    - Person will be asked to contribute fairly to next assignment and better manage workload 

## Project Timeline Proposal

| Meeting Date  | Meeting Time| Completed Before Meeting  | Discuss at Meeting |
|---|---|---|---|
| 4/28  |  4 PM | Brainstorm on project dataset/ question  | Determine best form of communication; Discuss and decide on project proposal; Assign work for project proposal | 
| 4/29  | 11 PM | Finish Assigned Work | Finishing touches, turn in Project proposal | 
| 5/05  | 6 PM  | Look into Checkpoint #1 | Assign individual work for Checkpoint #1 |
| 5/12  | 6 PM  | Work on Checkpoint #1 | Assign work for Checkpoint #1, Work together on Checkpoint #1 |
| 5/13  | 6 PM | Work on Checkpoint #1 | Finishing touches for Checkpoint #1, Turn in Checkpoint #1 |
| 5/18  | 3 PM  | Work on Checkpoint #2 | Assign work for Checkpoint #2, Work together on Checkpoint #2 |
| 5/25  | 3 PM  | Work on Checkpoint #2 | Work together on Checkpoint #2, Discuss any issues/conflicts |
| 5/27  | 11 PM | Work on Checkpoint #2 | Finishing touches for Checkpoint #2, Turn in Checkpoint #2 |
| 6/01  | 3 PM  | Look into Final project | Discuss what needs to be done and assign individual work for the final portion |
| 6/08  | 3 PM  | Work on Final Project | Touch base with team to see what progress needs to be made and help each other |
| 6/10  | 11 PM | Work on Final Project | Finishing touches for Final Project, Turn in Final Project |